### PR TITLE
✨Search page

### PR DESCRIPTION
**Pull request by joshbermanssw**
We created a Search Page!

Figma Mockup: https://www.figma.com/proto/Jyj0SszjfSBc6Minkl3g1l/Tina.io?node-id=1503-7209&node-type=frame&t=nFC6xWeRHd0cp8FV-0&scaling=scale-down-width&content-scaling=fixed&page-id=1503:6780&starting-point-node-id=1503:7209

**Changes:**
- [ ] Created new route /search which grabs query params from url (i.e /search?query=test) 

**PBIs created from this**
#2550 
#2551 
#2553

Screenshot 2024-11-29 at 12 21 38 pm

**Figure: Our new docs search page**

### CODE REVIEW

File: components/docsSearch/SearchComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+                    
+                    setIsFilterOpen(false);
+                  }}
+                >
+                  {option}
+                </div>
+              ))}
+            </div>
+          )}
+        </div>
```
Comment (isaaclombardssw) [inline components/docsSearch/SearchComponent.tsx:53]: Should this also be commented out?

File: components/docsSearch/SearchComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+          {/* Search Results Count */}
+          <div className="ml-auto text-end text-lg font-inter text-blue-800">
+            {numberOfResults}{' '}
+            Results
+          </div>
+        </div>
+        {isLoading && (
+          <div className="pt-10 text-2xl bg-gradient-to-br from-orange-300 via-orange-400 to-orange-600 bg-clip-text text-transparent font-tuner">
+            Gathering all the Llamas...
```
Comment (isaaclombardssw) [inline components/docsSearch/SearchComponent.tsx:157]: lol

File: components/docsSearch/SearchNavigation.tsx [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+            </div>
+          )
+        )}
+      </div>
+    </div>
+  );
+};
+
+export const SearchResultsOverflowTabs = ({ query }) => {
+  console.log('you passed me', query);
```
Comment (isaaclombardssw) [inline components/docsSearch/SearchNavigation.tsx:60]: Do we need this?

**Review by isaaclombardssw — APPROVED** [review]
LGTM

**Comment by Marxoz** [comment]
Once we display the search results, the preview results should disappear until a different search is entered 
 
image
Figure: Preview search results should disapear when we show the results page

**Review by Marxoz — CHANGES_REQUESTED** [review]
The old preview search results still pops up randomly

image
Figure: Old results preview still appears randomly

**Review by Marxoz — CHANGES_REQUESTED** [review]
@joshbermanssw  Once we display the search results, the preview results should disappear until a different search is entered 
 
image
Figure: Preview search results should disapear when we show the results page

File: components/docsSearch/SearchComponent.tsx (hunk 2 above)
Comment (bradystroud) [reply components/docsSearch/SearchComponent.tsx:157]:
```suggestion
            Mustering all the Llamas...
```

File: components/docsSearch/SearchNavigation.tsx (hunk 3 above)
Comment (joshbermanssw) [reply components/docsSearch/SearchNavigation.tsx:60]: removed

**Comment by joshbermanssw** [comment]
> @joshbermanssw Once we display the search results, the preview results should disappear until a different search is entered
> 
> image Figure: Preview search results should disapear when we show the results page

Done

File: components/docsSearch/SearchComponent.tsx (hunk 1 above)
Comment (joshbermanssw) [reply components/docsSearch/SearchComponent.tsx:53]: doesnt need to be :)

**Review by Marxoz — CHANGES_REQUESTED** [review]
On a smaller screen there is no search bar
image
Figure: Search bar missing.

**Comment by joshbermanssw** [comment]
> On a smaller screen there is no search bar image Figure: Search bar missing.

Part of the design - will consult with Betty re: Search Bar https://github.com/tinacms/tina.io/issues/2575

**Review by Marxoz — APPROVED** [review]

**Pull request merged**